lib/collection.js

```diff
@@ -0,0 +1,291 @@
+var Events = require('./events'),
+    Collection,
+    ARRAY_PROXY_METHODS = ['forEach', 'some', 'every', 'filter', 'map', 'reduce'];
+
+function clone(obj) {
```

Comment:
Object.create(obj)
But there are problem: properties of clone won't be enumerable
var clone = Object.create({a: {z: 1}, b: {z: 1}})
console.log(Object.keys(clone));// => []